rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbSaveToCustomCassandraTableNodeTest.java

```diff
@@ -0,0 +1,323 @@
+import org.thingsboard.server.dao.nosql.CassandraStatementTask;
+import org.thingsboard.server.dao.nosql.TbResultSet;
+import org.thingsboard.server.dao.nosql.TbResultSetFuture;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyDouble;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class TbSaveToCustomCassandraTableNodeTest {
+
+    private final DeviceId DEVICE_ID = new DeviceId(UUID.fromString("ac4ca02e-2ae6-404a-8f7e-c4ae31c56aa7"));
+    private final TenantId TENANT_ID = TenantId.fromUUID(UUID.fromString("64ad971e-9cfa-49e4-9f59-faa1a2350c6e"));
+
+    private final ListeningExecutor dbCallbackExecutor = new TestDbCallbackExecutor();
+
+    private TbSaveToCustomCassandraTableNode node;
+    private TbSaveToCustomCassandraTableNodeConfiguration config;
+
+    @Mock
+    private TbContext ctxMock;
+    @Mock
+    private CassandraCluster cassandraClusterMock;
+    @Mock
+    private GuavaSession sessionMock;
+    @Mock
+    private PreparedStatement preparedStatementMock;
+    @Mock
+    private BoundStatement boundStatementMock;
+    @Mock
+    private BoundStatementBuilder boundStatementBuilderMock;
+    @Mock
+    private ColumnDefinitions columnDefinitionsMock;
+    @Mock
+    private CodecRegistry codecRegistryMock;
+    @Mock
+    private ProtocolVersion protocolVersionMock;
+    @Mock
+    private Node nodeMock;
+
+    @BeforeEach
+    void setUp() {
+        node = spy(new TbSaveToCustomCassandraTableNode());
+        config = new TbSaveToCustomCassandraTableNodeConfiguration().defaultConfiguration();
+    }
+
+    @AfterEach
+    void tearDown() {
+        node.destroy();
+    }
+
+    @Test
+    void verifyDefaultConfig() {
+        assertThat(config.getTableName()).isEqualTo("");
+        assertThat(config.getFieldsMapping()).isEqualTo(Map.of("", ""));
+    }
+
+    @Test
+    void givenCassandraClusterIsMissing_whenInit_thenThrowsException() {
+        var configuration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        assertThatThrownBy(() -> node.init(ctxMock, configuration))
+                .isInstanceOf(RuntimeException.class)
```

Comment:
By default, TB tries to init the node 10 times before stopping the actor. For cases when the issue is unrecoverable, we use TbNodeException with unrecoverable flag set to true.

Could you please check if this issue might be recoverable? Try to stop the Cassandra cluster and after that initiate the node. After the certain number of retry attempts to start the cluster back. If the node will be started then change the exception to TbNodeException with unrecoverable flag set to false, otherwise set it to true.